doc/contributing/contributing.rst

```diff
@@ -31,49 +31,55 @@ To contribute to the source code of the model you will need to fork the reposito
 Quickstart Guide
 ----------------
 
-1. Fork the project on GitHub (using the fork button) and create a local clone, if you haven’t already:
+**1.** Fork the project on GitHub (using the fork button).
+
+**2.** Create a local clone (we strongly suggest keeping a separate repository for development work):
 
 ::
 
     % git clone https://github.com/user_name/MITgcm.git
 
-2. Move into your local clone directory (cd MITgcm) and and set up a remote that points to the original:
+**3.** Move into your local clone directory (cd MITgcm) and and set up a remote that points to the original:
 
 ::
 
     % git remote add upstream https://github.com/altMITgcm/MITgcm.git
 
-3. This step is optional, but highly recommended. It will ensure that your GitHub repo stays up to date with the main repository. Checkout the master branch and sync everywhere:
+**4.** Make a new branch from ``upstream/master`` (name it something appropriate, here we assume it is a new feature) and make edits on this branch:
 
 ::
 
-   % git checkout master
-   % git pull upstream master
-   % git push origin master
-   
-4. Make a new branch from ``upstream/master`` (name it something appropriate, here we assume it is a bugfix) and make edits on this branch:
+   % git fetch upstream
+   % git checkout -b newfeature upstream/master
 
-::
+**5.** When edits are done, do all git add’s and git commit’s. In the commit, make a succinct (<70 char) summary, followed by a blank line and a longer description. Reference any outstanding issues addressed using the syntax ``#ISSUE_NUMBER``.
 
-   % git fetch upstream
-   % git checkout -b bugfix upstream/master
+**6.** Push the edited branch to the origin remote (i.e. your fork) on GitHub:
 
-5. When edits are done, do all git add’s and git commit’s. In the commit, make a succinct (<70 char) summary, followed by a blank line and a longer description. Reference any outstanding issues addressed using the syntax ``#ISSUE_NUMBER``.
+::
 
+    % git push -u origin newfeature
 
-6. Push the edited branch to the origin remote (i.e. your fork) on GitHub:
+**7.** On GitHub, go to your fork and hit the pull request (PR) button, and wait for the MITgcm head developers to review your proposed changes. You may get additional questions or requests before your changes are accepted into the primary MITgcm source code.
+
+If you want to update your code branch before submitting (or any point in development), follow the following recipe. It will ensure that your GitHub repo stays up to date with the main repository. Note again that your edits should always be to a development branch (here, “newfeature”), not the master branch.
 
 ::
 
-    % git push -u origin bugfix
+    % git checkout master
+    % git pull upstream master
+    % git push origin master
```

Comment:
why is this necessary?

Reply:
Our git guinea pigs told us that they were confused when their fork didn't stay up to date with the main repo. These commands ensure their fork stays up to date. But, yes, they're not really necessary.

Reply:
That's a fair point. But I think that telling people to `push origin master` here creates false impression that it is necessary part of the process, which ultimately will cause greater confusion.

Once the fork has been made, origin/master never needs to be touched again. You merge directly from upstream/master to the local master.

Reply:
Your git guinea pigs and everyone else should be encouraged to *not* have a `master` branch in their development distro.  Then they won't be confused.  

They should not be checking out `master` they should be checking out `placeholder` or something anodyne.  They should periodically 

```
git fetch upstream master
git checkout placeholder
git reset --hard upstream/master
git checkout -b mypristinenewbranchoffmaster
```

This makes it crystal clear that `origin` is just a place for my changes, and doesn't mean anything except as a reference for PRs.  If you dont' do this, people are going to think `origin/master` is magically `upstream/master` somehow.

Reply:
http://matthew-brett.github.io/pydagogue/gh_delete_master.html

Reply:
I agree with you both, especially @rabernat's point about people potentially thinking `git push origin master` is a necessary part of the process; `origin/master` is not a branch that should ever be used. But, we asked people to tell us what they found difficult/off putting/strange about the new workflow and we got feedback from multiple people saying they were confused about why their repository wasn't staying up to date. Inserting a (somewhat superfluous) `git push origin master` is an easy way to avoid that confusion.

As much as I like the idea of telling people to remove `master` from their fork, I don't think those instructions are beginner friendly. Perhaps it would be better to think of this section of the manual as a "git survival guide for the uninitiated", rather than a description of the best possible workflow.

If you two are dead against having `git push origin master` in there, then I guess we'll have to keep working on this section.

Reply:
It's fine as is. We are splitting hairs.

Better to just get something up there and then iterate in the future if it seems necessary.